Tangra 3/VideoOperations/LightCurves/frmLightCurve.cs

```diff
@@ -1551,15 +1551,60 @@ private void miLoad_Click(object sender, EventArgs e)
 
         private void miSaveAsImage_Click(object sender, EventArgs e)
         {
-			if (m_Graph == null) return;
+            if (m_Graph == null) return;
+
             Bitmap bmp = AddImageLegend();
 
-			if (saveImageDialog.ShowDialog(this) == DialogResult.OK)
-			{
+            SetSaveAsImageFormatToLastUsed();
+
+            if (saveImageDialog.ShowDialog(this) == DialogResult.OK)
+            {
+                string file_ext = Path.GetExtension(saveImageDialog.FileName);
+
                 bmp.Save(
-					saveImageDialog.FileName,
-					GetImageFormatFromFileExtension(Path.GetExtension(saveImageDialog.FileName)));
-			}
+                    saveImageDialog.FileName,
+                    GetImageFormatFromFileExtension(file_ext));
+
+                SaveLastUsedSaveAsImageFormat(file_ext);
+            }
+        }
+
+        private void SetSaveAsImageFormatToLastUsed()
+        {
+
+            switch (TangraConfig.Settings.Generic.SaveAsImageFileType)
+            {
+                case TangraConfig.SaveAsImageFormat.Bmp:
```

Comment:
Changing the order of the filter is not really necessary. A better approach would be to use a single filter, but based on the saved setting to auto-suggest the file name with a different extension i.e. to set `saveImageDialog.FileName` before opening the dialog. The name of the file could be the name of the video files with the corresponding extension.